sdks/python/apache_beam/io/hadoopfilesystem_test.py

```diff
@@ -214,6 +214,11 @@ def setUp(self):
       url = self.fs.join(self.tmpdir, filename)
       self.fs.create(url).close()
 
+    try:                    # Python 2
```

Comment:
@aaltay , see related discussion here: https://issues.apache.org/jira/browse/BEAM-5620.

@HuangLED 
We could reduce the number of lines by checking `if sys.version_info[0] < 3:`, but I'm OK either way since this PR already adds value.

Reply:
@aaltay, sure.  just to clarify.   Did you mean move this to top of this Setup() function? 

@tvalentyn I might be wrong, but I think relying on sys.version_info[0] is more indirect way of verifying things, thus less preferred to me.  And in test file a bit code redundancy is not a bad thing. just my 2 cents.

Reply:
@HuangLED I think from code-readability standpoint, using `sys.version_info[0]` is a more direct way, since `sys.version_info` is a standard library function whose purpose is to return the version of interpreter. A reader of the code, who is not aware of `assertItemsEqual` deprecation will probably guess that this method is not available on Python 3.x, while `try/except` require a clarifying comment in order for a reader not to wonder why a method sometimes may not be available. Note that https://github.com/apache/beam/pull/6599/files is changing this as well, so consider removing these lies from the PR to avoid a merge conflict.

Reply:
Updated by moving the validation into SetUpClass, and relying on version_info.

Reply:
Thanks for sharing, @HuangLED . I'll link it in [quickstart guide](https://docs.google.com/document/d/1s1BJVCY65LB_SYK1SU1u7NbZiFANoq-nEYaEvzRbYlA). Using `sys.version_info[0] == 3` would definitely be a short-sighted check. I looked at Beam codebase, looks like we rely on it only to skip tests, in other places we use something like `sys.version_info[0] >= 3`, `sys.version_info[0] < 3`. Although feature detection is ultimately a recommended approach. Moving forward, we can use feature detection when we can, and add add a comment if necessary to explain why the check is needed. 
FYI @Fematich @Juta @manuzhang @RobbeSneyders @splovyt .